flink-java/src/main/java/org/apache/flink/api/java/operators/Keys.java

```diff
@@ -209,6 +209,9 @@ public ExpressionKeys(int[] groupingFields, TypeInformation<T> type, boolean all
 				throw new InvalidProgramException("Specifying keys via field positions is only valid " +
 						"for tuple data types. Type: " + type);
 			}
+			if (type.getArity() == 0) {
+				throw new InvalidProgramException("Tuple size must be greater than 0. Size: " + type.getArity());
```

Comment:
this check was added specifically for Tuple0, as in the current code you'd get weird exceptions when grouping on a Tuple0.